crates/weaver_live_check/src/attribute_advice.rs

```diff
@@ -0,0 +1,302 @@
+    fn advise(
+        &mut self,
+        attribute: &SampleAttribute,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Result<Vec<Advice>, Error>;
+}
+
+/// An advisor that checks if an attribute is deprecated
+pub struct DeprecatedAdvisor;
+impl Advisor for DeprecatedAdvisor {
+    fn advise(
+        &mut self,
+        _attribute: &SampleAttribute,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Result<Vec<Advice>, Error> {
+        let mut advices = Vec::new();
+        if let Some(attribute) = semconv_attribute {
+            if let Some(deprecated) = &attribute.deprecated {
+                advices.push(Advice {
+                    key: "deprecated".to_owned(),
+                    value: match deprecated {
+                        Deprecated::Renamed { .. } => Value::String("renamed".to_owned()),
+                        Deprecated::Obsoleted { .. } => Value::String("obsoleted".to_owned()),
+                        Deprecated::Uncategorized { .. } => {
+                            Value::String("uncategorized".to_owned())
+                        }
+                    },
+                    message: deprecated.to_string(),
+                    advisory: Advisory::Violation,
+                });
+            }
+        }
+        Ok(advices)
+    }
+}
+
+/// An advisor that checks if an attribute is stable from the stability field in the semantic convention
+/// The value will be the stability level
+pub struct StabilityAdvisor;
+// TODO: Configurable Advisory level, strictly stable would mean Violation
+
+impl Advisor for StabilityAdvisor {
+    fn advise(
+        &mut self,
+        _attribute: &SampleAttribute,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Result<Vec<Advice>, Error> {
+        let mut advices = Vec::new();
+        if let Some(attribute) = semconv_attribute {
+            match attribute.stability {
+                Some(ref stability) if *stability != Stability::Stable => {
+                    advices.push(Advice {
+                        key: "stability".to_owned(),
+                        value: Value::String(stability.to_string()),
+                        message: "Is not stable".to_owned(),
+                        advisory: Advisory::Improvement,
+                    });
+                }
+                _ => {}
+            }
+        }
+        Ok(advices)
+    }
+}
+
+/// An advisor that checks if an attribute has the correct type
+pub struct TypeAdvisor;
+impl Advisor for TypeAdvisor {
+    fn advise(
+        &mut self,
+        attribute: &SampleAttribute,
+        semconv_attribute: Option<&Attribute>,
+    ) -> Result<Vec<Advice>, Error> {
+        // Only provide advice if the attribute is a match and the type is present
+        match (semconv_attribute, attribute.r#type.as_ref()) {
+            (Some(semconv_attribute), Some(attribute_type)) => {
+                let semconv_attribute_type = match &semconv_attribute.r#type {
```

Comment:
I wonder if we should make a helper method on the struct of `Attribute` for this.  WDYT?

Reply:
Do you mean the conversion from the definition type to the accepted value type?